google-cloud-bigtable/samples/acceptance/read_samples_test.rb

```diff
@@ -101,9 +101,95 @@
 \tconnected_cell: 1 @#{@timestamp}
 \tconnected_wifi: 1 @#{@timestamp}
 \tos_build: PQ2A.190405.004 @#{@timestamp}
+
 OUTPUT
 
-    assert_match expected, out
+    assert_equal expected, out
+  end
+
+  it "reads_rows_nil" do
+    out, _err = capture_io do
+      reads_rows_nil bigtable_instance_id, @table_id
+    end
+    expected = <<~OUTPUT
+Reading data for phone#4c410523#20190501:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190405.003 @#{@timestamp}
+
+Reading data for phone#4c410523#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190405.004 @#{@timestamp}
+
+Reading data for phone#4c410523#20190505:
+Column Family stats_summary
+\tconnected_cell: 0 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190406.000 @#{@timestamp}
+
+Reading data for phone#5c10102#20190501:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190401.002 @#{@timestamp}
+
+Reading data for phone#5c10102#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 0 @#{@timestamp}
+\tos_build: PQ2A.190406.000 @#{@timestamp}
+
+Reading data for phone001#5c10102#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 0 @#{@timestamp}
+\tos_build: PQ2A.190406.000 @#{@timestamp}
+
+OUTPUT
+
+    assert_equal expected, out
+  end
+
+  it "reads_rows_empty" do
+    out, _err = capture_io do
+      reads_rows_empty bigtable_instance_id, @table_id
+    end
+    expected = <<~OUTPUT
+OUTPUT
+
+    assert_equal expected, out
+  end
+
+  it "reads_rows_sleep" do
+    Google::Cloud::Bigtable::V2::Bigtable::Client.configure do
+      # reads_rows_sleep is expected to return two elements
+      # let it timeout when processing 2nd element
+      # reproduce make sure its retry doesn't read extra (i.e. 3rd) element
+      config.rpcs.read_rows.timeout = 1.5
+    end
+
+    out, _err = capture_io do
+      reads_rows_sleep bigtable_instance_id, @table_id
+    end
+    expected = <<~OUTPUT
+Reading data for phone#4c410523#20190501:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190405.003 @#{@timestamp}
+
+Reading data for phone#4c410523#20190502:
+Column Family stats_summary
+\tconnected_cell: 1 @#{@timestamp}
+\tconnected_wifi: 1 @#{@timestamp}
+\tos_build: PQ2A.190405.004 @#{@timestamp}
+
```

Comment:
even though `reads_rows_sleep` only request two, turn out it actually returns more rows after `phone#4c410523#20190502`

https://github.com/googleapis/google-cloud-ruby/blob/535a7f85af1134ece4b84d46fe30b08ccf280f6c/google-cloud-bigtable/samples/read_samples.rb#L94-L97